src/theory/theory_model_builder.cpp

```diff
@@ -1177,7 +1177,7 @@ void TheoryEngineModelBuilder::debugCheckModel(TheoryModel* tm)
     for (; !eqc_i.isFinished(); ++eqc_i)
     {
       Node n = *eqc_i;
-      static int repCheckInstance = 0;
+      static thread_local int repCheckInstance = 0;
       ++repCheckInstance;
```

Comment:
For now that fix is OK, but I really don't like these random static variables in methods. @ajreynol do you have an idea why this is done this way?

Reply:
I would just delete this variable. Seems it was there only for debugging, but this can be deleted.

Reply:
Thanks! Any thoughts on the other ones?